src/cpprealm/app.cpp

```diff
@@ -515,9 +493,33 @@ namespace realm {
         client_config.user_agent_application_info = config.app_id;
 
         app_config.app_id = config.app_id;
-        app_config.transport = std::make_shared<internal::CoreTransport>(config.custom_http_headers, config.proxy_configuration);
         app_config.base_url = config.base_url;
 
+        // Sync socket provider configuration
+        if (config.sync_socket_provider) {
+            client_config.socket_provider = ::realm::internal::networking::create_sync_socket_provider_shim(config.sync_socket_provider,
+                                                                                                            config.proxy_configuration);
+        } else {
+            client_config.socket_provider = ::realm::internal::networking::create_sync_socket_provider_shim(std::make_shared<networking::default_socket_provider>(),
+                                                                                                            config.proxy_configuration);
```

Comment:
Instead of creating Core's default socket provider in our `default_socket_provider` and wrapping it with a shim why not just leave the `socket_provider` field empty if the user hasn't explicitly set one and let Core create its default socket provider directly?